huaweicloud/services/dli/resource_huaweicloud_dli_permission.go

```diff
@@ -0,0 +1,350 @@
+	if err != nil {
+		return nil, ParseDliErrorToError404(err)
+	}
+
+	if rst != nil && !rst.IsSuccess {
+		return nil, fmtp.Errorf("Error query DLI permission: %s", rst.Message)
+	}
+
+	for _, v := range rst.Privileges {
+		if v.UserName == userName {
+			return &v, nil
+		}
+	}
+
+	return nil, golangsdk.ErrDefault404{
+		ErrUnexpectedResponseCode: golangsdk.ErrUnexpectedResponseCode{
+			Body: []byte("Error query DLI permission"),
+		},
+	}
+}
+
+func resourceDliPermissionDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.DliV1Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud DLI v1 client: %s", err)
+	}
+
+	obj, userName := ParseTableInfoFromId(d.Id())
+	if strings.HasPrefix(obj, "queues.") {
+		queueName := strings.Replace(obj, "queues.", "", 1)
+		opts := auth.GrantQueuePermissionOpts{
+			QueueName:  queueName,
+			UserName:   userName,
+			Action:     "update",
+			Privileges: []string{},
+		}
+
+		rst, createErr := auth.GrantQueuePermission(client, opts)
+		if createErr != nil {
+			return fmtp.DiagErrorf("Error grant permission in DLI: %s", createErr)
+		}
+
+		if rst != nil && !rst.IsSuccess {
+			return fmtp.DiagErrorf("Error grant permission in DLI: %s", rst.Message)
+		}
+	} else {
+		opts := auth.GrantDataPermissionOpts{
+			UserName:   userName,
+			Action:     "update",
+			Privileges: []auth.DataPermission{{Object: obj, Privileges: []string{}}},
+		}
+
+		rst, createErr := auth.GrantDataPermission(client, opts)
+		if createErr != nil {
+			return common.CheckDeletedDiag(d, ParseDliErrorToError404(createErr), "DLI")
+		}
+
+		if rst != nil && !rst.IsSuccess {
+			return fmtp.DiagErrorf("Error delete DLI permission: %s", rst.Message)
+		}
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func ParseAuthInfoFromId(id string) (object, userName string) {
+	idArrays := strings.Split(id, "/")
+	object = idArrays[0]
+	userName = idArrays[1]
+	return
+}
+
+// Object format:
+// databases.Database_name
+// databases.Database_name.tables.Table_name
+// databases.Database_name.tables.Table_name.columns.Column_name
+// jobs.flink.Flink_job_ID
+// groups.Package_group_name
+// resources.PackageName
+// queues.queueName
+func QueryPermission(client *golangsdk.ServiceClient, obj, userName string) (*auth.Privilege, error) {
+	var permission *auth.Privilege
+	var err error
+
+	if strings.HasPrefix(obj, "databases") {
+		permission, err = queryDatabaseRelatePermission(client, obj, userName)
+	} else if checkPrefixMatchDataPermession(obj) {
+		permission, err = queryDataPermission(client, obj, userName)
+	} else if strings.HasPrefix(obj, "queues") {
+		permission, err = queryQueuePermission(client, obj, userName)
+	} else {
+		return nil, fmtp.Errorf("The object is illegal:object=%s,userName=%s, error=%s", obj, userName, err)
+	}
+
+	return permission, err
+}
+
+func ParseDliErrorToError404(respErr error) error {
```

Comment:
Will this function be used outside this package? If you don't use it, pls change to `parseDliErrorToError404`.

Reply:
yes,used